cinema/tests/tests_movie_view_set.py

```diff
@@ -0,0 +1,245 @@
+from cinema.serializers import MovieSessionSerializer
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+from django.db.models import F, Count
+
+from rest_framework.test import APIClient
+from rest_framework import status
+
+from cinema.models import Movie, MovieSession, CinemaHall, Genre, Actor
+from cinema.serializers import (
+    MovieSessionListSerializer,
+    MovieSessionSerializer,
+    MovieSessionDetailSerializer,
+)
+
+
+MOVIE_SESSION_LIST_URL = reverse("cinema:moviesession-list")
+
+
+def sample_movie_sessions(**params):
+    SCIFI = Genre.objects.create(name="Sci-Fi")
+    HORROR = Genre.objects.create(name="Horror")
+    ACTION = Genre.objects.create(name="Action")
+    COMEDY = Genre.objects.create(name="Comedy")
+
+    ARNY = Actor.objects.create(first_name="Arnold", last_name="Schwarzenegger")
+    LINDA = Actor.objects.create(first_name="Linda", last_name="Hamilton")
+    ALAN = Actor.objects.create(first_name="Alan", last_name="Tudyk")
+
+    MOVIE_1 = Movie.objects.create(
+        title="Terminator",
+        description="Robots, bang-bang",
+        duration=100,
+    )
+    MOVIE_1.actors.add(ARNY, LINDA)
+    MOVIE_1.genres.add(SCIFI, ACTION)
+
+    MOVIE_2 = Movie.objects.create(
+        title="Resident Alien",
+        description="Shoes, chavk-chavk",
+        duration=10,
+    )
+    MOVIE_2.actors.add(LINDA, ALAN)
+    MOVIE_2.genres.add(SCIFI, COMEDY)
+
+    RED = CinemaHall.objects.create(name="Red", rows=10, seats_in_row=10)
+    BLUE = CinemaHall.objects.create(name="Blue", rows=12, seats_in_row=12)
+
+    MovieSession.objects.create(
+        show_time="2024-10-08 13:00:00", movie=MOVIE_1, cinema_hall=RED
+    )
+    MovieSession.objects.create(
+        show_time="2024-10-09 13:00:00", movie=MOVIE_2, cinema_hall=BLUE
+    )
+
+
+class UnauthenticatedUserTest(TestCase):
+    def setup(self):

```

Comment:
The method `setup` should be renamed to `setUp` to correctly override the `TestCase` method for setting up test fixtures. This ensures that the setup code is executed before each test method.